lib/pdk/cli/exec/command.rb

```diff
@@ -0,0 +1,233 @@
+require 'bundler'
+require 'childprocess'
+require 'English'
+require 'tempfile'
+require 'tty-spinner'
+require 'tty-which'
+
+require 'pdk/util'
+require 'pdk/util/git'
+require 'pdk/util/ruby_version'
+
+module PDK
+  module CLI
+    module Exec
+      class Command
+        attr_reader :argv
+        attr_reader :context
+        attr_accessor :timeout
+        attr_accessor :environment
+        attr_writer :exec_group
+
+        def initialize(*argv)
+          @argv = argv
+
+          @process = ChildProcess.build(*@argv)
+          @process.leader = true
+
+          @stdout = Tempfile.new('stdout').tap { |io| io.sync = true }
```

Comment:
Warning: Tempfile doesn't set binary mode, so CRLF will be used on Windows which may not be what you want.

Reply:
The stuff in `lib/pdk/cli/exec/command.rb` was extracted out into it's own file but pretty much unchanged from the current implementation. So this is a good thing to know but shouldn't be a regression from current behavior.